src/main/java/com/cloudbees/jenkins/support/impl/ProcFilesRetriever.java

```diff
@@ -45,6 +46,8 @@ protected static String getNodeName(Node node) {
      */
     abstract public Map<String, String> getFilesToRetrieve();
 
+    abstract protected List<Node> getNodes();
```

Comment:
My only concern is that now all existing implementors of ProcFilesRetriever will fail as they don't implement `getNodes`, but I suppose it's not very used out there. A mention on the **Changelog** talking about this loss of compatibility is a must-have, I think. cc: @varyvol

Reply:
makes sense @MRamonLeon (limited risk due to the number of existing extensions)

Reply:
I missed the compatibility concern completely because I was just trying to fix asap and continue with other work :(
Do you think there is a way to make it backward compatible? I can probably make it concrete and return a list including the master and all the nodes.

Reply:
Good point @MRamonLeon I totally missed it.

@duemir probably you can make the method in the ProcFilesRetriever class not be abstract but return both the Jenkins instance and the nodes (previous behaviour) and then you override it in the subclasses?

Reply:
Done. Please let me know what you think.